From: Andrew Ruder <andrew.ruder@elecsyscorp.com>
Subject: [PATCH] enable cross compilation for expect

This patch was created by running ./configure on a modern Linux machine
and inserting the results into the cross compilation section of
each AC_MSG_CHECKING that bombed out with an error.

Signed-off-by: Andrew Ruder <andrew.ruder@elecsyscorp.com>
---

Index: expect-5.45/configure.in
===================================================================
--- expect-5.45.orig/configure.in	2013-11-14 07:59:58.732100595 -0600
+++ expect-5.45/configure.in	2013-11-14 07:59:58.732100595 -0600
@@ -481,7 +481,7 @@
 ,
 	AC_MSG_RESULT(no)
 ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+	AC_MSG_RESULT(no)
 )
 
 AC_MSG_CHECKING([if any value exists for WNOHANG])
@@ -506,7 +506,8 @@
 	AC_MSG_RESULT(no)
 	AC_DEFINE(WNOHANG_BACKUP_VALUE, 1)
 ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+	AC_MSG_RESULT(yes)
+	AC_DEFINE(WNOHANG_BACKUP_VALUE, 1)
 )
 
 #
@@ -574,7 +575,7 @@
 	AC_DEFINE(REARM_SIG)
 ,
 	AC_MSG_RESULT(no)
-, AC_MSG_WARN([Expect can't be cross compiled])
+, AC_MSG_RESULT(no)
 )
 
 # HPUX7 has trouble with the big cat so split it
@@ -725,7 +726,7 @@
 ,
         AC_MSG_RESULT(no)
 ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+	AC_MSG_RESULT(no)
 )
 
 # mach systems have include files for unimplemented features
@@ -749,7 +750,9 @@
 ,
         AC_MSG_RESULT(no)
 ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+    AC_DEFINE(HAVE_TERMIO)
+    PTY_TYPE=termios
+	AC_MSG_RESULT(yes)
 )
 
   # now check for the new style ttys (not yet posix)
@@ -771,7 +774,9 @@
   ,
         AC_MSG_RESULT(no)
   ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+        AC_DEFINE(HAVE_TERMIOS)
+        PTY_TYPE=termios
+        AC_MSG_RESULT(yes)
   )
 fi
 
@@ -794,7 +799,7 @@
 ,
 	AC_MSG_RESULT(no)
 ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+	AC_MSG_RESULT(no)
 )
 
 AC_MSG_CHECKING([if TIOCGWINSZ in termios.h])
@@ -816,7 +821,7 @@
 ,
 	AC_MSG_RESULT(no)
 ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+	AC_MSG_RESULT(no)
 )
 
 # finally check for Cray style ttys
@@ -837,7 +842,7 @@
 ,
 	AC_MSG_RESULT(no)
 ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+	AC_MSG_RESULT(no)
 )
 
 #
@@ -869,7 +874,9 @@
 AC_HAVE_FUNCS(getpty)
 
 # following test sets SETPGRP_VOID if setpgrp takes 0 args, else takes 2
-AC_FUNC_SETPGRP
+cat >>expect_cf.h <<\_ACEOF
+#define SETPGRP_VOID 1
+_ACEOF
 
 #
 # check for timezones
@@ -889,7 +896,7 @@
 	AC_MSG_RESULT(yes),
 	AC_MSG_RESULT(no)
 ,
-	AC_MSG_ERROR([Expect can't be cross compiled])
+	AC_MSG_RESULT(yes)
 )
 
 
